dashboard/app/models/sections/section.rb

```diff
@@ -112,6 +134,22 @@ def participant_type_not_changed
     end
   end
 
+  # We want the `grades` attribute to be a list that only includes elements
+  # that exist in the list of valid grades.
+  def grades_are_subset_of_valid_grades
+    unless Section.valid_grades?(grades)
+      errors.add(:grades, "must be one or more of the valid student grades. Expected: #{VALID_GRADES}. Got: #{grades}.")
+    end
+  end
+
+  # If the grades include 'pl', they must *ONLY* include 'pl'.
+  # E.g.: You can't have a section with 'K' and 'pl'.
+  def grades_with_pl_are_only_pl
+    if grades.include?(SharedConstants::PL_GRADE_VALUE) && grades.length != 1
```

Comment:
Should this also check that this is a pl section?

Reply:
there was an existing validation for that, so I thought it'd be better to keep them separate